docker-compose.yml

```
@@ -214,12 +221,20 @@ services:
       - "8126:8126"
 
   # https://hub.docker.com/_/elasticsearch/
-  # Note: This image is deprecated, but the new one requires fiddling.
-  elasticsearch:
+  legacy-elasticsearch:
     image: mozilla/socorro_elasticsearch:1.4.5
     ports:
       - "9200:9200"
-      - "9300:9300"
+
+  # https://www.elastic.co/guide/en/elasticsearch/reference/8.14/docker.html
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:8.14.3
+    command:
+    - bin/elasticsearch
+    - -Expack.security.enabled=false
+    - -Ediscovery.type=single-node
```

Nit: These need to be indented two more spaces.

```suggestion
      - bin/elasticsearch
      - -Expack.security.enabled=false
      - -Ediscovery.type=single-node
```

---

Also, in the current configuration, I think elasticsearch 8 checks the environment to see how much memory it can take and then _takes all the memory_. On my laptop, running the elasticsearch container and anything else causes my laptop to run out of memory and then a bunch of processes get killed (usually Slack and some others).

If I do `docker-compose up legacy-elasticsearch`, it doesn't take up much memory--I don't notice it in the performance monitor. It also doesn't log much. There's nothing about heap sizes in the logs.

If I do `docker-compose up elasticsearch`, it immediately takes 15gb+ of memory and my system gets sad. If I look at the logs, it's _very chatty_. In the extraordinary amount of output, I see this line:

```
socorro-elasticsearch-1         | {"@timestamp":"2024-08-28T15:24:05.701Z", "log.level": "INFO", "message":"JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -Djava.security.manager=allow, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Dlog4j2.formatMsgNoLookups=true, -Djava.locale.providers=SPI,COMPAT, --add-opens=java.base/java.io=org.elasticsearch.preallocate, --add-opens=org.apache.lucene.core/org.apache.lucene.store=org.elasticsearch.vec, --enable-native-access=org.elasticsearch.nativeaccess, -Des.cgroups.hierarchy.override=/, -XX:ReplayDataFile=logs/replay_pid%p.log, -Djava.library.path=/usr/share/elasticsearch/lib/platform/linux-x64:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib, -Djna.library.path=/usr/share/elasticsearch/lib/platform/linux-x64:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib, -Des.distribution.type=docker, -XX:+UnlockDiagnosticVMOptions, -XX:G1NumCollectionsKeepPinned=10000000, -XX:+UseG1GC, -Djava.io.tmpdir=/tmp/elasticsearch-6759128300942634220, --add-modules=jdk.incubator.vector, -XX:+HeapDumpOnOutOfMemoryError, -XX:+ExitOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,level,pid,tags:filecount=32,filesize=64m, -Xms15882m, -Xmx15882m, -XX:MaxDirectMemorySize=8329887744, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=25, --module-path=/usr/share/elasticsearch/lib, --add-modules=jdk.net, --add-modules=ALL-MODULE-PATH, -Djdk.module.main=org.elasticsearch.server]", "ecs.version": "1.2.0","service.name":"ES_ECS","event.dataset":"elasticsearch.server","process.thread.name":"main","log.logger":"org.elasticsearch.node.Node","elasticsearch.node.name":"fac4b151d2b3","elasticsearch.cluster.name":"docker-cluster"}
```

The interesting bit I think is this:

```
-Xms15882m, -Xmx15882m
```

I checked what we were doing when we were trying to migrate to elasticsearch 7:

5a7480001736a2d5c0b484520651e2255cfa1311

We don't set any memory params there. Maybe elasticsearch 7 did different things with heap sizes? Maybe we weren't actually running the container so I never got to find out?

Anyhow, for the local dev environment, I think we need to cap the heap size. If I add these params, then everything works fine for me:

```
-Xms2g
-Xmx2g
```

I picked those values mostly to test an example in the advanced configuration docs. I don't know if those are appropriate values.

https://www.elastic.co/guide/en/elasticsearch/reference/current/advanced-configuration.html#set-jvm-heap-size

Also, we might want to specify some logging related configuration to reduce the amount of chattiness and make it more human readable. It's in a structured format currently.

https://www.elastic.co/guide/en/elasticsearch/reference/current/logging.html

---

per https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html
> 5. Start an Elasticsearch container.
> ```
> docker run --name es01 --net elastic -p 9200:9200 -it -m 1GB docker.elastic.co/elasticsearch/elasticsearch:8.15.0
> ```
> > Use the -m flag to set a memory limit for the container. This removes the need to [manually set the JVM size](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-set-heap-size).

equivalent option in docker compose is [`mem_limit`](https://docs.docker.com/reference/compose-file/services/#mem_limit), which I have set to `1g` to match the above example